ratpack-session/src/main/java/ratpack/session/internal/DefaultSessionKey.java

```diff
@@ -19,12 +19,16 @@
 import ratpack.api.Nullable;
 import ratpack.session.SessionKey;
 
+import java.util.Objects;
+
 public class DefaultSessionKey<T> implements SessionKey<T> {
```

Comment:
I don't have enough context here, but doesn't unintentionally having a `DefaultSessionKey(null, null)` potentially introduce a security risk in itself?

Reply:
That state isn't possible. I've moved the check to make that clearer.

I can't see how it would be a risk. There'd be a runtime error before trying to do anything with it.